osquery/tables/system/posix/extended_attributes.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/posix/xattr_utils.h"
+
+#include <osquery/tables.h>
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace {
+Status appendDirectoryEntryAttributes(QueryData& results,
+                                      const std::string& path) {
+  ExtendedAttributes attributes;
+  if (!getExtendedAttributes(attributes, path)) {
+    return Status(
+        1,
+        "Failed to acquire the extended attributes for the following path: " +
+            path);
+  }
+
+  for (const auto& p : attributes) {
+    const auto& name = p.first;
+    const auto& value = p.second;
+
+    std::string parent_path;
+    try {
+      parent_path = boost::filesystem::path(path).parent_path().string();
+    } catch (...) {
```

Comment:
Can this not be more specific?

Reply:
I thought it would throw an exception in case of error but I was wrong! I'm fixing it now